### ISSUE TITLE
Correctly extract associated function in trait item

### ISSUE THREAD

**Issue by Undin** [intellij-rust/intellij-rust#2185]
We can extract associated function in `impl Ty { ... }` block but not for trait item.

```rust
trait Foo {
    fn foo(&self) {
        let b = 1;
        <selection>println!("{}", b);</selection>
    }
}
```

**Expected**:
```rust
trait Foo {
    fn foo(&self) {
        let b = 1;
        Self::bar(b);
    }

    fn bar(b: i32) {
        println!("{}", b);
    }
}
```

**Actual**:
```rust
trait Foo {
    fn foo(&self) {
        let b = 1;
        bar(b);  // error[E0425]: cannot find function `bar` in this scope
    }

    fn bar(b: i32) {
        println!("{}", b);
    }
}
```

Also, looks like we can use `Self` type for all associated function extraction not only in trait item to unify implementation of these cases

**Comment by t-kameyama** [comment]
I will work on this issue.

**Comment by Undin** [comment]
Fixed in #2194
